### PR TITLE
repo_mirror init

### PR DESCRIPTION
**Pull request by fujexo**
The role isn't yet perfect and still has some flaws but can you have an initial look? We can improve it afterwards, this is more or less the working copy from pkg.adfinis-sygroup.ch

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,15 +1,15 @@
-===============
-ROLE _TEMPLATE
-===============
+================
+ROLE repo_mirror
```
Comment (keachi) [inline README.rst]: Use uppercase and `s/_/ /g`

File: defaults/main.yml [hunk 2]
```diff
@@ -1 +1,137 @@
...
+
+# Deploy the fedora mirror report script
+repo_mirror_fedora_report: false
+repo_mirror_fedora_report_name: '<name>'
+repo_mirror_fedora_report_pass: '<password>'
+
+# A list of fedora projects to report. Default is an empty list.
+repo_mirror_fedora_reports: []
+
+#  - name: 'Fedora EPEL'
```
Comment (keachi) [inline defaults/main.yml]: Add
```
# .. code-block:: YAML
#
#   repo_mirror_fedora_reports:
```
before, so the YAML to RST convertion will look correctly.

File: defaults/main.yml [hunk 3]
```diff
@@ -1 +1,137 @@
...
+#    path: '/var/www/mirror/epel'
+#  - name: 'Fedora Linux'
+#    enabled: False
+#    path: '/var/www/mirror/fedora'
+
+# A list of repositories. Default is an empty list.
+# Below are some examples how to define a repository to mirror
+repo_mirror_repos: []
+
+#  - name: alpine
```
Comment (keachi) [inline defaults/main.yml]: See `repo_mirror_fedora_reports` comment

File: defaults/main.yml [hunk 4]
```diff
@@ -1 +1,137 @@
...
+#    cron_hour: '*'
+#    cron_minute: '07'
+#    cron_disabled: true # optional
+#
+#  - name: ubuntu
+#    type: rsync_ubuntu
+#    source_repo: rsync://ch.rsync.archive.ubuntu.com/ubuntu
+#    cron_hour: '*/2'
+#    cron_minute: '41'
+#    cron_disabled: true # optional
```
Comment (keachi) [inline defaults/main.yml]: Fix your editor, there is no newline at the end of the file.

File: meta/main.yml [hunk 5]
```diff
@@ -18,8 +18,13 @@ galaxy_info:
...
     - name: EL
       versions:
         - 6
         - 7
-  galaxy_tags: []
+  galaxy_tags:
+    - repository
+    - mirror
+    - rsync
+    - wget
```
Comment (keachi) [inline meta/main.yml]: Fix your editor.

File: tasks/configuration.yml [hunk 6]
```diff
@@ -1 +1,109 @@
...
+    mode: 0740
+  with_items: "{{ repo_mirror_repos }}"
+  when: item is defined and item
+  tags:
+    - 'role::repo_mirror'
+    - 'role::repo_mirror:configuration'
+    - 'role::repo_mirror:configuration:script'
+
+- name: create log rotation config
+  template:
```
Comment (keachi) [inline tasks/configuration.yml]: Also specify the user, group and mode.

File: meta/main.yml [hunk 7]
```diff
@@ -4,7 +4,7 @@ dependencies: []
 
 galaxy_info:
```
Comment (karras) [inline meta/main.yml]: Please add `role_name: repo_mirror` to ensure galaxy imports it correctly.

Comment (fujexo) [reply meta/main.yml]: Done

**Pull request merged**